Source/WebKit/_WebKit_SwiftUI/Foundation+Extras.swift

```diff
@@ -35,16 +35,55 @@ func onNextMainRunLoop(do body: @escaping @MainActor () -> Void) {
 }
 
 struct AnyEquatable: Equatable {
-    let value: Any
-    private let equals: (Any) -> Bool
+    private enum Storage {
+        case value(Any, (Any) -> Bool)
+        case never
+    }
+
+    private let storage: Storage
+
+    var value: Any {
+        switch storage {
+        case let .value(value, _): value
+        case .never: ()
+        }
+    }
 
     init<E: Equatable>(_ value: E) {
-        self.value = value
-        self.equals = { ($0 as! E) == value }
+        self.storage = .value(value, { other in
+            guard let other = other as? E else {
+                assertionFailure()
+                return false
+            }
+
+            return other == value
+        })
+    }
+
+    /// Creates an `AnyEquatable` that is never equal to anything.
+    init() {
+        self.storage = .never
     }
 
-    static func == (lhs: AnyEquatable, rhs: AnyEquatable) -> Bool {
-        lhs.equals(rhs.value)
+    static func == (left: AnyEquatable, right: AnyEquatable) -> Bool {
+        switch (left.storage, right.storage) {
+        case (.never, _), (_, .never):
+            false
+
+        case let (.value(leftValue, leftEquals), .value(rightValue, rightEquals)):
+            leftEquals(rightValue) && rightEquals(leftValue)
+        }
+    }
+}
+
+@MainActor
+func withContinuousObservation<T>(of value: @MainActor @escaping @autoclosure () -> T, execute: @MainActor @escaping (T) -> Void) {
```

Comment:
Doesn’t matter _now_, but I wonder if you could make this generic over `any Actor` or somehow take in the isolation so isn’t `MainActor`-only.